src/infra/core/src/ingest/polling_ingest_service_impl.rs

```diff
@@ -140,17 +140,17 @@ impl PollingIngestServiceImpl {
         &self,
         args: IngestLoopArgs,
     ) -> Result<PollingIngestResult, PollingIngestError> {
-        let ctx = ingest_common::new_session_context(self.object_store_registry.clone());
-
-        let mut data_writer = DataWriterDataFusion::builder(args.dataset.clone(), ctx.clone())
-            .with_metadata_state_scanned(None)
-            .await
-            .int_err()?
-            .build();
-
         let mut iteration = 0;
         let mut combined_result = None;
+
         loop {
+            let ctx = ingest_common::new_session_context(self.object_store_registry.clone());
+            let mut data_writer = DataWriterDataFusion::builder(args.dataset.clone(), ctx.clone())
+                .with_metadata_state_scanned(None)
```

Comment:
During the initialization `DataWriter` needs to restore the state of processing from metadata. This requires finding a lot of events like polling source (for merge strategy), vocabulary (for system column names), etc. - very often this will result in a full-scan of the metadata chain.

Reusing writer across iterations allows it to incrementally update its internal state, minimizing scanning - I'd hate to lose this optimization.

Please hold off on this change and let me take a closer took at Datafusion API to see if there is a better way. All we really need is to restore the context state to a point before any views were registered in it for processing or clear the state entirely.

Reply:
Done, proposed another solution approach